doc/api/v0.yaml

```diff
@@ -615,10 +615,55 @@ paths:
           content:
             text/plain:
               schema:
-                description: Hex-encoded ID of the posted transaction
-                type: string
-                pattern: '[0-9a-f]+'
-              example: 68dcc12fe0dfe5e7b66ca6f8c959f9aa43b273e120a77fc3e4e2f04f1ecd7968
+                description: The information about whether a message was accepted or rejected
+                type: object
+                properties:
+                  accepted:
+                    description: IDs of accepted fragments
+                    type: array
+                    items:
+                      type: string
+                      format: '[0-9a-f]+'
+                  rejected:
+                    description: Detailed information about rejected fragments
+                    type: array
+                    items:
+                      type: object
+                      required:
+                        - id
+                        - reason
+                      properties:
+                        id:
+                          description: The ID of a rejected fragment
+                          type: string
+                          format: '[0-9a-f]+'
+                        reason:
+                          oneOf:
+                            - description: This fragment was already processed by this node
+                              type: string
+                              enum:
+                                - FragmentAlreadyInLog
```

Comment:
I think a cleaner approach, that is less confusing to OpenAPI tools, would be to make `reason` a discriminator field, but then you can't probably make one of its variant values an object unto itself (it's more typical in JSON to grow extension fields alongside a particular value of the discriminator field; but it's probably not how serde-json would map a Rust enum).

Any examples of using `enum` have a unified value list under the field type. Also see the [JSON schema spec draft](https://tools.ietf.org/html/draft-bhutton-json-schema-validation-00#section-6.1.2).

Reply:
I do not like that hack as well and I think it is ugly. There is not much we can do about that, since we have multiple mempools for different leaders and we need to report the exact mempool for which the problem appeared.

Another thing we can do to (it would result into basically the same API): unify all errors except `PoolOverflow` under one enum and leave `PoolOverflow` as it is now.

Reply:
Also, JSON schema is pretty neat, but would require a lot of work from us :)

Reply:
> Also, JSON schema is pretty neat, but would require a lot of work from us :)

Well, if we want to have a neat OpenAPI spec, JSON schema is what it is based on.

Reply:
@eugene-babichenko So in the OpenAPI spec we stick all scalar variants into a single `enum`, define non-scalar variants as object types, and merge all this under some combination of `allOf`/`anyOf` for the containing object?

Reply:
Yes, pretty much that.